textproc/lhs2tex/Portfile

```diff
@@ -1,22 +1,31 @@
 # -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
 
 PortSystem          1.0
-PortGroup           haskell 1.0
+PortGroup           haskell_stack 1.0
 
-haskell.setup       lhs2tex 1.18.1 ghc no
-name lhs2tex
-revision            3
-checksums           rmd160  047072143ddce8e83239a2e77c0f4fc6a2425132 \
-                    sha256  dccae22231a1db64cddc38b6f7ac98ac4d5677f03df2f548b2efe9a5e63e9648
+name                lhs2tex
+version             1.22
+checksums           rmd160  257e58b02249fd8403b14d314a519552afd2a5ab \
+                    sha256  7eea3b5b4bae019aa9acb6036e6c2606c62a0d9969a081d4b9b9f04baa3129bd \
+                    size    679988
 
 license             GPL-2+
 maintainers         {cal @neverpanic} openmaintainer
 platforms           darwin
-
+categories          textproc tex haskell
 description         Preprocessor for typesetting Haskell sources with LaTeX
 long_description    ${description}
 
-depends_lib-append  port:hs-mtl \
-                    port:hs-regex-compat \
-                    port:texlive-basic \
-                    port:texlive-latex-extra
+homepage            https://hackage.haskell.org/package/${name}
+master_sites        ${homepage}-${version}
+
+pre-build {
```

Comment:
Should we add an option to the haskell_stack PortGroup that does this guarded by another option?

Reply:
Yes, it should go in the portgroup. I suspect we can intelligently determine if it's necessary by looking for (the absence of) `stack*.yaml` files, but I was [hoping](https://github.com/macports/macports-ports/pull/5050#issuecomment-527283457) to get @essandess's feedback on that idea.

Reply:
@amake @neverpanic I think this is a good idea, with an option like `use_stack_init yes`.

I believe the right thing to do is not add a `pre-build` block, but change the `configure.args` to be `init`, rather than `setup`, and the subsequent calls to `stack`, I believe will take care of the rest.

I’ve looked at the `stack`  guide https://docs.haskellstack.org/en/stable/README/ to see if you’re supposed to say `stack init` then `stack setup` Or vice-versa, but I don’t think it matters.

As for `stack.yaml`, that should be a simple tcl call to see if it exists. The other `stack-*.yaml` files are for builds with previous versions. You may want to add a `--stack-yaml` option to specify which file is used. Or just keep it simple. Your call.

@amake Would you please try the suggestion of changing `configure.opts` to say `init` and see if that works?

Reply:
Also, we should have an automated way of checking whether cabal is hardcoding a `datadir` with `${worksrcpath}` into the binaries of the new `stack` ports so that the port can enter the joy of the discussion at PR https://github.com/macports/macports-ports/pull/5180 of how to fix `cabal`‘s lack of destdir.

@amake I suggest you check these by hand for now before they’re merged.